packages/web-components/src/badge/badge.ts

```diff
@@ -44,29 +43,4 @@ export class Badge extends FASTBadge {
    */
   @attr
   public size: BadgeSize = 'medium';
-
-  /**
-   *
-   * Default slotted content
-   *
-   * @public
-   * @remarks
-   */
-  public defaultSlottedContent: HTMLElement[];
-
-  /**
-   * Applies 'icon-only' class when there is only an SVG in the default slot
-   *
-   * @public
-   * @remarks
-   */
-  public defaultSlottedContentChanged(): void {
-    const slottedElements = this.defaultSlottedContent.filter(x => x.nodeType === Node.ELEMENT_NODE);
-
-    if (slottedElements.length === 1 && slottedElements[0] instanceof SVGElement) {
-      this.classList.add('icon-only');
-    } else {
-      this.classList.remove('icon-only');
-    }
-  }
```

Comment:
Two changes here:

1. I couldn't figure out what `defaultSlottedContent` was doing that including an empty `<slot>` element couldn't do. So I've reduced this to an empty `<slot>` in the template.
2. The icon-only logic is implicit and probably is not possible to write properly for every use case. If this is required, I'd propose we go for an explicit and declarative attr if required so the user can take on the responsibility of the logic behind when to add/remove it.
3. I couldn't find icon-only in the specs or React implementation, so I was also curious if this might have been copy pasta / cruft from all the churn.

LMK

Reply:
Looks like this removes (incidentally) the size - should we keep that attr?

Reply:
Great call out - I don't think this is even necessary for badge; this was in _button_...I think you're correct with a copy/pasta  issue. Ultimately, I agree here that even in button we'd likely look to go with just an explicit attribute and use that as the styling hook...if we go the ElementInternals route we could also have it be a state...either way I think we have a better approach all-up here, but this for sure isn't used here.